src/main/java/org/opensearch/security/auditlog/impl/AbstractAuditLog.java

```diff
@@ -628,7 +633,7 @@ public void logDocumentWritten(ShardId shardId, GetResult originalResult, Index
             }
         }
 
-        if (!complianceConfig.shouldLogWriteMetadataOnly()) {
+        if (!complianceConfig.shouldLogWriteMetadataOnly() && !complianceConfig.shouldLogDiffsForWrite()) {
```

Comment:
Add this condition in so that the full request body is only logged if both `compliance.write_metadata_only` and `compliance.write_log_diffs` are both `false`. These settings seems to be in conflict:

Documentation: https://opensearch.org/docs/latest/security/access-control/api/

```
compliance.write_log_diffs	Boolean	Logs only diffs for document updates. Default is false.

compliance.write_metadata_only	Boolean	Log only metadata of the document for write events. Default is true.
```

The full body should only get audit logged if both `compliance.write_metadata_only` and `compliance.write_log_diffs` are `false`.